userspace/libsinsp/event.h

```diff
@@ -451,8 +451,70 @@ class SINSP_PUBLIC sinsp_evt : public gen_event
 
 		uint32_t nparams = scap_event_decode_params(m_pevt, params);
 
+		/* We need the event info to overwrite some parameters if necessary. */
+		const struct ppm_event_info* event_info = &m_event_info_table[m_pevt->type];
+		int param_type = 0;
+		
 		for(j = 0; j < nparams; j++)
 		{
+			/* Here we need to manage a particular case:
+			* 
+			*    - PT_CHARBUF
+			*    - PT_FSRELPATH
+			*    - PT_BYTEBUF
+			*    - PT_BYTEBUF
+			* 
+			* In the past these params could be `<NA>` or `(NULL)` or empty.
+			* Now they can be only empty! The ideal solution would be:
+			* 	params[i].buf = NULL;
+			*	params[i].size = 0;
+			* 
+			* The problem is that userspace is not
+			* able to manage `NULL` pointers... but it manages `<NA>` so we
+			* convert all these cases to `<NA>` when they are empty!
+			* 
+			* If we read scap-files we could face `(NULL)` params, so also in
+			* this case we convert them to `<NA>`.
+			* 
+			* To be honest there could be another corner case, but right now
+			* we don't have to manage it:
+			*    
+			*    - PT_SOCKADDR
+			*    - PT_SOCKTUPLE
+			*    - PT_FDLIST
+			* 
+			* Could be empty, so we will have:
+			* 	params[i].buf = "pointer to the next param";
+			*	params[i].size = 0;
+			* 
+			* However, as we said in the previous case, the ideal outcome would be:
+			* 	params[i].buf = NULL;
+			*	params[i].size = 0;
+			* 
+			* The difference with the previous case is that the userspace can manage
+			* these params when they have `params[i].size == 0`, so we don't have
+			* to use the `<NA>` workaround! We could also introduce the `NULL` and so
+			* put in place the ideal solution for this parameter, but before doing this
+			* we need to be sure that the userspace never tries to deference the pointer
+			* otherwise it will trigger a segmentation fault at run-time. So as a first
+			* step we would keep them as they are.
+			*/
+			param_type = event_info->params[j].type;
+			
+			if((param_type == PT_CHARBUF ||
+				param_type == PT_FSRELPATH ||
+				param_type == PT_FSPATH)
+				&&
+				(params[j].size == 0 ||
+				(params[j].size == 7 && strncmp((char*)params[j].buf, "(NULL)", 7) == 0)))
+			{
+				/* Overwrite the value and the size of the param.
+				* 5 = strlen("<NA>") + `\0`.
+				*/
+				params[j].buf = (void*)"<NA>";
+				params[j].size = 5;
+			}
```

Comment:
This seems the right place to put the conversion logic, we could put it also in `scap_event_decode_params` but having it in sinsp seems clearer to me, and allow us to test the event output a different levels (scap and sinsp)